plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/SqlSupplier.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import java.sql.SQLException;
+
+@FunctionalInterface
+public interface SqlSupplier<T>
```

Comment:
For now it can be a private interface in `LazyConnectionFactory`. We can make it public & reusable when needed